### PR TITLE
With NDBC and Argo runs and other changes

### PR DESCRIPTION
**Pull request by GwenChen-NOAA**
## Pull Request Testing ##

- [X] Describe testing already performed for this Pull Request: Run in cron for couples of months and successfully generate stat files. Proof-of-concept approved by EVS team leads. </br>

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions: 1) copy /lfs/h2/emc/ptmp/Lichuan.Chen/evs into your /lfs/h2/emc/ptmp/$USER directory. This directory contains pre-processed data files from previous 12 days needed for a run. Otherwise, you need to run the ecf files for at least 11 days and stat files will be generated after 10 days; 2) clone the EVS repo into your noscrub directory at /lfs/h2/emc/vpppg/noscrub; 3) qsub jevs_rtofs_prep.ecf, qsub jevs_rtofs_stats.ecf, qsub jevs_rtofs_ndbc_stats.ecf, qsub jevs_rtofs_plots.ecf, and qsub jevs_rtofs_headline_plots.ecf to run. Run jevs_rtofs_prep.ecf before jevs_rtofs_stats.ecf. Run jevs_rtofs_stats.ecf between 22:00 and 23:45 UTC to make sure data are available for the rundate. Run jevs_rtofs_plots.ecf after jevs_rtofs_stats.ecf. In this version of code, a copy of the stat files will be archived at /lfs/h2/emc/vpppg/noscrub/$USER/stat_archive/RTOFS. This is needed for testing purpose. This function will be removed in the final code. Thanks! </br>

- [X] Has the code been checked to ensure that no errors occur during the execution? Yes.

- [X] Do these updates/additions include sufficient testing updates? Yes.

- [X] Please complete this pull request review by 04/10/2023.</br>

## Pull Request Checklist ##

- [X] Review the source issue metadata (required labels, projects, and milestone).
- [X] Complete the PR description above.
- [X] Ensure the PR title matches the feature branch name.
- [X] Check the following:
- [X]  Instructions provided on how to run
- [X]  Developer's name is replaced by ${user} where necessary throughout the code
- [X]  Check that the ecf file has all the proper definitions of variables
- [X]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [X]  Check to see that the output directory structure is followed
- [X]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [X] After submitting the PR, select **Development** issue with the original issue number.
- [X] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [X] Close the linked issue.

### CODE REVIEW

File: jobs/rtofs/stats/JEVS_RTOFS_NDBC_STATS
Comment (PerryShafran-NOAA) [inline jobs/rtofs/stats/JEVS_RTOFS_NDBC_STATS:1]: Realistically, there should be only one JOBS file.  Can you combine this into the main JOBS file?

File: ecf/rtofs/stats/jevs_rtofs_ndbc_stats.ecf
Comment (PerryShafran-NOAA) [inline ecf/rtofs/stats/jevs_rtofs_ndbc_stats.ecf:1]: Is there a reason why the ndbc stats ecf file is separate from the rest of the rtofs ecf file?

**Comment by GwenChen-NOAA** [comment]
Yes, it is because ndbc stats job takes 3+ hours to run by itself, so I
separated it from the rest.

Gwen

**Comment by PerryShafran-NOAA** [comment]
Hi, Gwen,

In the prep output file /lfs/h2/emc/vpppg/noscrub/perry.shafran/EVS_gwen_rtofs_ndbc/ecf/rtofs/prep/jevs_rtofs_prep.o51323151, I'm getting a lot of Permission denied statements.  Can you look at this file and see why that might be?  I don't think I've gotten that before.

I have your prep output here:  /lfs/h2/emc/ptmp/perry.shafran/evs/v1.0/prep/rtofs

Thanks!

Perry

**Comment by GwenChen-NOAA** [comment]
Interesting!  I don't have those permission denied messages in my log files
(e.g., /lfs/h2/emc/vpppg/noscrub/Lichuan.Chen/EVS/ecf/rtofs/prep/jevs_rtofs_prep.o51258622).
Could it be a quota issue?  But it only happened to the *hvr*.nc files.
Those files are not used in EVS.  I delete them in
the EVS/scripts/rtofs/prep/exevs_rtofs_prep.sh script.

The prep output
at /lfs/h2/emc/ptmp/perry.shafran/evs/v1.0/prep/rtofs/rtofs.20230322 looks
good to me.

Thanks,
Gwen

**Comment by PerryShafran-NOAA** [comment]
Interesting.  Look throughout the file, there are lots of files in which there is a permission denied issue.  Good that the prep data looks good, but this would likely raise some red flags for NCO.  I'll do the stats and plots for now, but we'll have to look into this.

Perry

**Comment by GwenChen-NOAA** [comment]
I can modify my script to be more precise and only copy the needed files.
Do you want me to do that?

Gwen

**Comment by PerryShafran-NOAA** [comment]
That might help, but it interesting that even though there are permission denied statements the files they are working on exist in the directory.  Maybe it's because I had already copied in your run?  I'm letting it run offline in my own account over the weekend, and I'll let you know.

Thanks!

Perry

**Comment by GwenChen-NOAA** [comment]
Sounds good, thanks!

Gwen

**Comment by PerryShafran-NOAA** [comment]
OK, good news.  The latest run doesn't have the permission errors.  I'll check again on Monday and let you see the results then.

Perry

**Comment by PerryShafran-NOAA** [comment]
Hi, @GwenChen-NOAA ,

You may now validate all the prep, stats, and plots in the /lfs/h2/emc/ptmp/perry.shafran/evs/v1.0 directory.  Look for date 20230324.

Thanks!

Perry

**Comment by GwenChen-NOAA** [comment]
Perry,

Your stat files of valid date 20230324 are identical to mine, and the prep
directories look fine.  Time series and mean vs. lead plots are generated
as desired except that they only contain 3 days of data.
Performance diagrams are missing for the OSISAF run
(/lfs/h2/emc/ptmp/perry.shafran/evs/v1.0/plots/rtofs/rtofs.20230324/osisaf).
I'm running my jevs_rtofs_plots.ecf job to see if there is a bug in my code.

Thanks,
Gwen

**Comment by GwenChen-NOAA** [comment]
Perry,

I do have performance diagrams generated in my job.  Please
see /lfs/h2/emc/ptmp/Lichuan.Chen/evs/v1.0/plots/rtofs/rtofs.20230325/osisaf.
CSI plots are also missing in your run.  Please check.

Thanks,
Gwen

**Comment by PerryShafran-NOAA** [comment]
I fixed the issue by bringing in all the stat files from your archive.  Please take a look here:  /lfs/h2/emc/ptmp/perry.shafran/evs/v1.0/plots/rtofs/rtofs.20230324.

Thanks!

Perry

**Comment by GwenChen-NOAA** [comment]
Thanks, Perry!  The new plots look very good.

Gwen

**Comment by PerryShafran-NOAA** [comment]
Excellent! I can now go ahead and merge this PR.  Soon, I'll have the parallel PR for you to look over and approve.

Perry

**Review by PerryShafran-NOAA — APPROVED** [review]

**Pull request merged**